### PR TITLE
Updated 'subnet' entity with search and update mixin

### PR DESCRIPTION
**Pull request by sghai**
Updated `Subnet` entity with search and update mixin.

`Search results:`
```
In [8]: entities.Subnet(server_config, name="testsubnet", network="192.168.142.0", mask="255.255.255.0").create()
Out[8]: nailgun.entities.Subnet(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), dns_primary=None, domain=[], name=u'testsubnet', organization=[], dns_secondary=None, id=2, network=u'192.168.142.0', boot_mode=u'DHCP', from=None, ipam=u'DHCP', mask=u'255.255.255.0', vlanid=None, gateway=None, to=None, location=[], dns=None, dhcp=None, tftp=None)

In [9]: subnet = entities.Subnet(server_config).search(query={u'search': u'network="192.168.142.0"'})

In [10]: subnet
Out[10]: [nailgun.entities.Subnet(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), dns_primary=None, dns_secondary=None, gateway=None, network=u'192.168.142.0', name=u'testsubnet', from=None, ipam=u'DHCP', mask=u'255.255.255.0', vlanid=None, id=2, to=None, dns=None, dhcp=None, tftp=None, boot_mode=u'DHCP')]
```

### CODE REVIEW

**Comment by sghai** [comment]
Please note that I've added `update` method explicitly because server was not returning the full description while updating the `dns` `dhcp` fields via updatemixin only.

```
In [26]: subnet = subnet.update(['dns'])
WARNING:nailgun.client:Received HTTP 500 response: {
  "error": {"message":"dns_id is not allowed as nested parameter for subnets. Allowed parameters are domain_id, location_id, organization_id"}
}
```

**Comment by sghai** [comment]
```
In [5]: subnet = entities.Subnet(server_config).search(query={u'search': u'network="192.168.142.0"'})[0]
In [6]: proxy = entities.SmartProxy(server_config).search(query={u'search': u'name="foreman-server"'})[0]

In [20]: subnet.dns = proxy
In [22]: subnet.dhcp = proxy

subnet = subnet.update(['dns', 'dhcp'])

In [27]: subnet.dhcp
Out[27]: nailgun.entities.SmartProxy(nailgun.config.ServerConfig(url='foreman-server', verify=False, auth=('admin', 'changeme')), id=1)
```

File: nailgun/entities.py [hunk 1]
```diff
@@ -3397,6 +3402,25 @@ def read(self, entity=None, attrs=None, ignore=None):
         ignore.add('discovery')
         return super(Subnet, self).read(entity, attrs, ignore)
 
+    def update_payload(self, fields=None):
+        """Wrap submitted data within an extra dict."""
+        return {
```
Comment (oshtaier) [inline nailgun/entities.py]: As far as I see, it can be one line statement:
`return {u'subnet': super(Subnet, self).update_payload(fields)}`

Comment (sghai) [inline nailgun/entities.py]: :+1:

**Comment by abalakh** [comment]
ACK

File: nailgun/entities.py [hunk 2]
```diff
@@ -3397,6 +3402,20 @@ def read(self, entity=None, attrs=None, ignore=None):
...
 
+    def update_payload(self, fields=None):
+        """Wrap submitted data within an extra dict."""
+        return {u'subnet': super(Subnet, self).update_payload(fields)}
+
+    def update(self, fields=None):
+        """Fetch a complete set of attributes for this entity.
+
+        For more information, see `Bugzilla #1250922
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1250922>`_.
```
Comment (Ichimonji10) [inline nailgun/entities.py]: Is this method necessary? I can completely drop this `update` method and sucessfully update a subnet. Also, the linked-to bug is entitled "PUT /api/compute_resources/:id returns an incomplete set of attributes". If this method is necessary, it should link to a relevant bug.

**Comment by Ichimonji10** [comment]
ACK pending comment.

File: nailgun/entities.py (hunk 2 above)
Comment (sghai) [inline nailgun/entities.py]: Ok..let me retry without `update` method.

Comment (sghai) [inline nailgun/entities.py]: Yes.. that works without `update`. thank you

**Comment by sghai** [comment]
Test results after removing `update` method.
```
In [5]: proxy = entities.SmartProxy(server_config).search(query={u'search': u'name="foreman_server"'})[0]
In [6]: subnet = entities.Subnet(server_config).search(query={u'search': u'network="192.168.143.0"'})[0]

In [7]: subnet.tftp = proxy

In [8]: subnet = subnet.update(['tftp'])

In [9]: subnet.tftp
Out[9]: nailgun.entities.SmartProxy(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=1)
```

**Comment by Ichimonji10** [comment]
Thank you. ACK.

**Comment by oshtaier** [comment]
ACK

**Comment by elyezer** [comment]
:+1:

**Pull request merged**